src/lib/utils/breadcrumbs.ts

```diff
@@ -0,0 +1,69 @@
+import { BreadcrumbItem } from '@/types/dataTypes/drupal/field_type'
+
+export function deriveLastBreadcrumbFromPath(
+  breadcrumbs: BreadcrumbItem[],
+  breadcrumbTitle: string,
+  currentPath: string,
+  replaceLastItem: boolean
+): BreadcrumbItem[] {
+  const last: BreadcrumbItem = {
+    uri: currentPath,
+    title: breadcrumbTitle,
+    options: [],
+  }
+
+  if (replaceLastItem) {
+    breadcrumbs.splice(breadcrumbs.length - 1, 1, last)
+  } else {
+    breadcrumbs.push(last)
+  }
+
+  return breadcrumbs
+}
+
+// Intended for handling breadcrumbs for learning center content types
+export function deriveLcBreadcrumbs(
+  breadcrumbs: BreadcrumbItem[],
+  breadcrumbTitle: string,
+  currentPath: string,
+  lcBreadcrumbsTitleInclude?: boolean
+): BreadcrumbItem[] {
+  const filteredCrumbs: BreadcrumbItem[] = breadcrumbs.filter(
+    (crumb) => crumb.uri !== '/resources'
+  )
+
+  filteredCrumbs.push({
+    uri: currentPath,
+    title: 'Resources and support',
+    options: [],
+  })
+
+  if (lcBreadcrumbsTitleInclude) {
+    filteredCrumbs.push({
+      uri: currentPath,
+      title: breadcrumbTitle,
+      options: [],
+    })
+  }
+
+  return filteredCrumbs
+}
+
+function getRelativePathFromURL(uri) {
+  return new URL(uri, 'http://va.gov').pathname
+}
```

Comment:
This looks great, though I think it will be most useful if it's pulled out (and exported from) a separate file. Maybe `@/lib/utils/url` or something like that.